cmd/iop/mandiff.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iop
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/operator/pkg/manifest"
+	"istio.io/pkg/log"
+)
+
+type manDiffArgs struct {
+	// compareDir indicates comparison between directory.
+	compareDir bool
+}
+
+func manifestDiffCmd(rootArgs *rootArgs, diffArgs *manDiffArgs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "mandiff",
+		Short: "Compare manifests and generate diff.",
+		Long:  "The mandiff subcommand is used to compare manifest from two files or directories.",
+		Args:  cobra.ExactArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			if diffArgs.compareDir {
+				compareManifestsFromDirs(args[0], args[1])
+			} else {
+				compareManifestsFromFiles(rootArgs, args)
+			}
+		}}
+	return cmd
+}
+
+func addManDiffFlag(cmd *cobra.Command, diffArgs *manDiffArgs) {
+	cmd.PersistentFlags().BoolVarP(&diffArgs.compareDir, "directory", "r",
+		false, "compare directory")
+}
+
+//compareManifestsFromFiles compares two manifest files
+func compareManifestsFromFiles(rootArgs *rootArgs, args []string) {
+	if err := configLogs(rootArgs); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Could not configure logs: %s", err)
+		os.Exit(1)
+	}
+	a, err := ioutil.ReadFile(args[0])
```

Comment:
First check that you have the correct number of input files - this could crash. 
You'd always want two files or dirs (although a stdin option would also be nice).

Reply:
yep I actually had the cobra.ExactArgs(2) enforcement at line 40